libs/gl-client/src/signer/mod.rs

```diff
@@ -281,31 +289,25 @@ impl Signer {
         }
     }
 
-    fn authenticate_request(&self, req: &HsmRequest) -> Result<Vec<Approval>, Error> {
-        let ctxrequests: Vec<model::Request> = self
-            .check_request_auth(req.requests.clone())
-            .into_iter()
-            .filter_map(|r| r.ok())
-            .map(|r| decode_request(r))
-            .filter_map(|r| match r {
-                Ok(r) => Some(r),
-                Err(e) => {
-                    log::error!("Unable to decode request in context: {}", e);
-                    None
-                }
-            })
-            .collect::<Vec<model::Request>>();
+    fn authenticate_request(
+        &self,
+        msg: &vls_protocol::msgs::Message,
+        reqs: &Vec<model::Request>,
+    ) -> Result<(), Error> {
+        log::trace!(
+            "Resolving signature request against pending grpc commands: {:?}",
+            reqs
+        );
 
-        use auth::Authorizer;
-        let auth = auth::GreenlightAuthorizer {};
-        let approvals = auth.authorize(ctxrequests).map_err(|e| Error::Auth(e))?;
-        debug!("Current approvals: {:?}", approvals);
-        Ok(approvals)
+        // Quick path out of here: we can't find a resolution for a
+        // request, then abort!
+        Resolver::try_resolve(msg, &reqs)?;
+
+        Ok(())
```

Comment:
`Resolver::try_resolve` has the same return type as `authenticate_request` itself. We may obay returning an explicit `Ok(())` and instead return the result of `Resolve::try_resolve` directly.
```suggestion
        Resolver::try_resolve(msg, &reqs)?
```

Reply:
... without the `?` operator

Reply:
Thanks @devrandom, changed the suggestion.